File: Java-Core/src/main/java/prompto/utils/CmdLineParser.java
package prompto.utils;

import java.util.HashMap;
import java.util.Map;

import org.antlr.v4.runtime.ANTLRInputStream;
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.tree.ParseTree;
import org.antlr.v4.runtime.tree.ParseTreeProperty;
import org.antlr.v4.runtime.tree.ParseTreeWalker;

import prompto.utils.ArgsLexer;
import prompto.utils.ArgsParser;
import prompto.utils.ArgsParserBaseListener;
import prompto.utils.ArgsParser.ELEMENTContext;
import prompto.utils.ArgsParser.EntryContext;
import prompto.utils.ArgsParser.KeyContext;
import prompto.utils.ArgsParser.STRINGContext;

public class CmdLineParser {

	public static Map<String,String> parse(String input) throws Exception {
		if(input==null)
			input = "";
		try {
			CharStream stream = new ANTLRInputStream(input);
			ArgsLexer lexer = new ArgsLexer(stream);
			CommonTokenStream tokens = new CommonTokenStream(lexer);
			ArgsParser parser = new ArgsParser(tokens);
			ParseTree tree = parser.parse();
			CmdLineBuilder builder = new CmdLineBuilder();
			ParseTreeWalker walker = new ParseTreeWalker();
			walker.walk(builder, tree);
			return builder.getCmdLineArgs();
		} catch (RecognitionException e) {
			e.printStackTrace(System.err);
			throw new Exception(e);
		}
	}
	
	static class CmdLineBuilder extends ArgsParserBaseListener {
		
		Map<String,String> args = new HashMap<String,String>();
		ParseTreeProperty<String> values = new ParseTreeProperty<String>();
		
		public Map<String, String> getCmdLineArgs() {
			return args;
		}

		@Override
		public void exitEntry(EntryContext ctx) {
			String key = values.get(ctx.k);
			String value = values.get(ctx.v);
			args.put(key,value);
		}

		@Override
		public void exitKey(KeyContext ctx) {
			values.put(ctx,ctx.getText());
		}

		@Override
		public void exitSTRING(STRINGContext ctx) {
			String s = ctx.getText();
			values.put(ctx,s.substring(1,s.length()-1));
		}

		@Override
		public void exitELEMENT(ELEMENTContext ctx) {
			values.put(ctx,ctx.getText());
		}
		 
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/utils/CmdLineParser.js
var antlr4 = require("antlr4/index");
var al = require("./ArgsLexer");
var ap = require("./ArgsParser");
var ArgsParserListener = require("./ArgsParserListener").ArgsParserListener;

function CmdLineBuilder() {
	ArgsParserListener.call(this);
	this.args = {};
	this.values = {};
	return this;
}

CmdLineBuilder.prototype = Object.create(ArgsParserListener.prototype);
CmdLineBuilder.prototype.constructor = CmdLineBuilder;

CmdLineBuilder.prototype.getCmdLineArgs = function() {
	return this.args;
};

CmdLineBuilder.prototype.exitEntry = function(ctx) {
	var key = this.values[ctx.k];
	var value = this.values[ctx.v];
	this.args[key] = value;
};

CmdLineBuilder.prototype.exitKey = function(ctx) {
	this.values[ctx] = ctx.getText();
};

CmdLineBuilder.prototype.exitSTRING = function(ctx) {
	var s = ctx.getText();
	this.values[ctx] = s.substring(1,s.length-1);
};

CmdLineBuilder.prototype.exitELEMENT = function(ctx) {
	this.values[ctx] = ctx.getText();
};

function CmdLineParser() {
	return this;
}

CmdLineParser.parse = function(input) {
	if(input===null) {
		input = "";
	}
	var stream = new antlr4.InputStream(input);
	var lexer = new al.ArgsLexer(stream);
	var tokens = new antlr4.CommonTokenStream(lexer);
	var parser = new ap.ArgsParser(tokens);
	var tree = parser.parse();
	var builder = new CmdLineBuilder();
	var walker = new antlr4.tree.ParseTreeWalker();
	walker.walk(builder, tree);
	return builder.getCmdLineArgs();
};

exports.CmdLineParser = CmdLineParser;